tests/unit/amo/components/TestSearchPage.js

```diff
@@ -165,6 +166,39 @@ describe(__filename, () => {
     sinon.assert.notCalled(fakeDispatch);
   });
 
+  it('dispatches a server redirect when `platform` parameter is "all"', () => {
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    render({ location: { query: { platform: 'all' } }, store });
+
+    sinon.assert.calledWith(fakeDispatch, sendServerRedirect({
+      status: 302,
+      url: '/en-US/android/search/',
+    }));
+    sinon.assert.callCount(fakeDispatch, 1);
+  });
+
+  it('does not dispatch a server redirect when `platform` is not "all"', () => {
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    render({ location: { query: { platform: OS_MAC } }, store });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('redirects without affecting the other parameters', () => {
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+    const query = { atype: 1, page: 123, platform: 'all' };
```

Comment:
Since `atype` has its own mapping logic, this would be a more neutral test without it. If the query were set to `{ page: 123, platform: 'all' }` then I think the test coverage would be just as good.

Reply:
Yeah I wanted to test both at once, `atype` (which has a dedicated logic) and `page` (which is yet another parameter). I've removed `atype`.